micrometer-core/src/main/java/io/micrometer/core/instrument/MeterRegistry.java

```diff
@@ -258,49 +257,55 @@ public Search tags(String... tags) {
             return tags(zip(tags));
         }
 
-        public Search value(Statistic statistic, double value) {
-            valueAsserts.put(statistic, value);
-            return this;
-        }
-
-        public Optional<Timer> timer() {
-            return findAny(Timer.class);
+        public Timer timer() {
+            return findOne(Timer.class);
         }
 
-        public Optional<Counter> counter() {
-            return findAny(Counter.class);
+        public Counter counter() {
+            return findOne(Counter.class);
         }
 
-        public Optional<Gauge> gauge() {
-            return findAny(Gauge.class);
+        public Gauge gauge() {
+            return findOne(Gauge.class);
         }
 
-        public Optional<FunctionCounter> functionCounter() {
-            return findAny(FunctionCounter.class);
+        public FunctionCounter functionCounter() {
+            return findOne(FunctionCounter.class);
         }
 
-        public Optional<TimeGauge> timeGauge() {
-            return findAny(TimeGauge.class);
+        public TimeGauge timeGauge() {
+            return findOne(TimeGauge.class);
         }
 
-        public Optional<FunctionTimer> functionTimer() {
-            return findAny(FunctionTimer.class);
+        public FunctionTimer functionTimer() {
+            return findOne(FunctionTimer.class);
         }
 
-        public Optional<DistributionSummary> summary() {
-            return findAny(DistributionSummary.class);
+        public DistributionSummary summary() {
+            return findOne(DistributionSummary.class);
         }
 
-        public Optional<LongTaskTimer> longTaskTimer() {
-            return findAny(LongTaskTimer.class);
+        public LongTaskTimer longTaskTimer() {
+            return findOne(LongTaskTimer.class);
         }
 
-        private <T> Optional<T> findAny(Class<T> clazz) {
-            return meters()
+        private <T> T findOne(Class<T> clazz) {
+            Optional<T> meter = meters()
                 .stream()
-                .filter(m -> clazz.isInstance(m))
+                .filter(clazz::isInstance)
                 .findAny()
                 .map(clazz::cast);
+
+            if(meter.isPresent()) {
+                return meter.get();
+            }
+
+            String tagDetail = "";
+            if (!tags.isEmpty()) {
+                tagDetail = " with Tags:[" + tags.stream().map(t -> t.getKey()+":"+t.getValue()).collect( Collectors.joining( "," ) ) + "]";
+            }
+
+            throw new AssertionError("Unable to locate a meter named '"+name+"'"+tagDetail+" of type "+clazz);
```

Comment:
Figured I'd take a crack at the 'ideal' approach. I removed the `Optional` and throw a descriptive error.

This also allows removing the `map` method call. I really like it.